sw/device/lib/dif/dif_aes.c

```diff
@@ -0,0 +1,424 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_aes.h"
+
+#include <stddef.h>
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+
+#include "aes_regs.h"  // Generated.
+
+/*
+ * From: https://docs.opentitan.org/hw/ip/aes/doc/index.html#register-table,
+ * aes.CTRL.
+ */
+
+/*
+ * Field to select AES block cipher mode.
+ */
```

Comment:
It seems you have accidentally removed the comment explaining the values and resolution of invalid values. Do you want to add that back?

Reply:
Done.